staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/managedfieldsupdater.go

```diff
@@ -72,11 +72,23 @@ func (f *managedFieldsUpdater) Apply(liveObj, appliedObj runtime.Object, managed
 	if err != nil {
 		return object, managed, err
 	}
+
 	if object != nil {
+		// non-nil object means the apply operation modified the object, so update the manager's timestamp
 		managed.Times()[fieldManager] = &metav1.Time{Time: time.Now().UTC()}
 	} else {
+		// nil object means the apply operation did not modify the input object
+		// clone the input object and return it without managed fields
 		object = liveObj.DeepCopyObject()
 		RemoveObjectManagedFields(object)
+
+		if _, managerInFields := managed.Fields()[fieldManager]; managerInFields {
+			if _, managerInTimes := managed.Times()[fieldManager]; !managerInTimes {
+				// if the manager owns fields, ensure it has an associated time.
+				// a no-op apply can add co-ownership of existing field values, so record the time that occurred.
+				managed.Times()[fieldManager] = &metav1.Time{Time: time.Now().UTC()}
```

Comment:
@jpbetz for existing managed entries that got created without timestamps, once this bug fix releases, the *next* no-op update will set the time to `now` on a pre-existing managed field entry. That's a little weird, and probably ok, but I wanted to call it out explicitly as a side-effect of the fix.

Reply:
We fixed a few spurious resourceVersion bumps issues this year:

- https://github.com/kubernetes/kubernetes/pull/125317
- https://github.com/kubernetes/kubernetes/pull/125263

So I'm keeping an eye out of anything that changes an apply request *before* the object is compared with the stored resource to decide if a write is needed. But this happens *after*  `fieldmanager.Apply()` so appears safe, at least from that perspective.